src/transformations/InsertFinalNewline.ts

```diff
@@ -23,22 +24,59 @@ export default class InsertFinalNewline extends PreSaveTransformation {
 		const lineCount = doc.lineCount;
 		const lastLine = doc.lineAt(lineCount - 1);
 
-		if (!editorconfigProperties.insert_final_newline
-			|| lineCount === 0
-			|| lastLine.isEmptyOrWhitespace) {
+		if (lineCount === 0) {
 			return { edits: [] };
 		}
 
-		const position = new Position(
-			lastLine.lineNumber,
-			lastLine.text.length
-		);
+		if (editorconfigProperties.insert_final_newline) {
+			if (lastLine.isEmptyOrWhitespace) {
+				return { edits: [] };
+			} else {
```

Comment:
You've already returned early in the "if" block above, so there's no reason to use an else here. Please apply the same paradigm throughout the PR.